src/lib/schematics/nav/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.ts

```diff
@@ -16,13 +16,25 @@ import { map } from 'rxjs/operators';
   encapsulation: ViewEncapsulation.<%= viewEncapsulation %><% } if (changeDetection !== 'Default') { %>,
   changeDetection: ChangeDetectionStrategy.<%= changeDetection %><% } %>
 })
-export class <%= classify(name) %>Component {
+export class <%= classify(name) %>Component implements OnDestroy {
 
-  isHandset$: Observable<boolean> = this.breakpointObserver.observe(Breakpoints.Handset)
-    .pipe(
-      map(result => result.matches)
-    );
+  isHandset: boolean;
 
-  constructor(private breakpointObserver: BreakpointObserver) {}
+  /** Emits when the component is destroyed. */
+  private readonly _destroyed = new Subject<void>();
+
+  constructor(breakpointObserver: BreakpointObserver) {
+    breakpointObserver.observe(Breakpoints.Handset)
+      .pipe(
+        takeUntil(this._destroyed)
+      ).subscribe(result => {
```

Comment:
Can we do something like:

```ts
breakpointObserver.observe(Breakpoints.Handset)
  .pipe(takeUntil(this._destroyed))
  .subscribe(result => this.isHandset = result.matches);
```

Basically the same code, just a bit more compact and more readable IMO.

Reply:
I don't mind removing spaces, but `{}` make sense to me.. It signifies that the arrow function does not return a value and has a side-effect.

Would this be OK?:
```
breakpointObserver.observe(Breakpoints.Handset)
  .pipe(takeUntil(this._destroyed))
  .subscribe(result => {this.isHandset = result.matches;});
```